src/nvim/statusline.c

```diff
@@ -647,6 +634,10 @@ int build_stl_str_hl(win_T *wp, char *out, size_t outlen, char *fmt, int use_san
     stl_separator_locations = xmalloc(sizeof(int) * stl_items_len);
   }
 
+  // if "fmt" was set insecurely it needs to be evaluated in the sandbox
+  const int use_sandbox = opt_name ? was_set_insecurely(wp, opt_name, opt_scope)
```

Comment:
API function `nvim_eval_statusline()` will not pass along an option name. Opted to pass along NULL instead and null check here. This patch didn't end up porting as cleanly as I would have hoped but I guess it's still an improvement.

Reply:
I think you can add a comment here about it.